python_ta/transforms/type_inference_visitor.py

```diff
@@ -318,20 +322,25 @@ def visit_return(self, node):
     def visit_functiondef(self, node):
         arg_types = [self.type_constraints.lookup_concrete(node.type_environment.lookup_in_env(arg))
                      for arg in node.argnames()]
-
-        # Check whether this is a method in a class
-        if isinstance(node.parent, astroid.ClassDef) and isinstance(arg_types[0], TypeVar):
-            self.type_constraints.unify(arg_types[0], _ForwardRef(node.parent.name))
-
-        # check if return nodes exist; there is a return statement in function body.
-        if len(list(node.nodes_of_class(astroid.Return))) == 0:
-            func_type = Callable[arg_types, None]
+        if any(annotation is not None for annotation in node.args.annotations):
+            func_type = parse_annotations(node)
+            [self.type_constraints.unify(arg_type, annotation) for arg_type
+                , annotation in zip(arg_types, func_type.__args__[:-1])]
```

Comment:
Don't break lines before commas. But also don't use comprehensions when you are evaluating an expression just for a side effect. In such cases, use a loop.